hooks/tk-multi-publish2/basic/publish_session.py

```diff
@@ -0,0 +1,712 @@
+# Copyright (c) 2017 Shotgun Software Inc.
+# 
+# CONFIDENTIAL AND PROPRIETARY
+# 
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit 
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your 
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import os
+import pprint
+import maya.cmds as cmds
+import maya.mel as mel
+import sgtk
+from sgtk.util.filesystem import copy_file
+import traceback
+
+HookBaseClass = sgtk.get_hook_baseclass()
+
+
+class MayaSessionPublishPlugin(HookBaseClass):
```

Comment:
this is perhaps a studid question, but did you consider having one set of plugins for 'auto' and one for templates? I can see pros and cons with both, and ultimately i guess the design of the collector may play into this decision too. Just wanted to point it out!

Reply:
Having a single set of hooks was called out in the ticket by @robblau so wanted to give this a shot first. This is why I didn't go too much farther with this first pass since I wanted to make sure this approach looks reasonable to everyone. I think we need to continue to think about hook complexity, code duplication, as well while we're trying to decide what to do here. Let's discuss.

Reply:
Having used this a bit while testing, it feels nice having the fallback logic in there. I will outline the workflow in a comment on the PR to help guide the discussion.